source/Halibut/Diagnostics/LogCreators/AggregateLogWriterLogCreator.cs

```diff
@@ -0,0 +1,22 @@
+using System;
+using Halibut.Diagnostics.LogWriters;
+
+namespace Halibut.Diagnostics.LogCreators
+{
+    public class AggregateLogWriterLogCreator : ICreateNewILog
+    {
+        readonly ICreateNewILog nonCachingLogFactory;
+        readonly Func<string, ILogWriter[]> logWriterFactoryForPrefix;
+
+        public AggregateLogWriterLogCreator(ICreateNewILog nonCachingLogFactory, Func<string, ILogWriter[]> logWriterFactoryForPrefix)
```

Comment:
Nit: I know the func makes it more reusable, but looking at the example in the Background, gosh it'd be nice to have it with `params`.

Maybe we could add a second constructor like this `public AggregateLogWriterLogCreator(ICreateNewILog nonCachingLogFactory, params ILogWriter[] logWriters)`, and all it does is call the other one with a special func that returns just those objects.

I think that Octopus Server would only use the same instances anyway, right?

Then the example in Background would look like this:

```
These log writers only record and can never return logs.
ILogWriter logConnectionErrors = new LogConnectionErrors();
ILogWriter logSecurityEvents = new LogSecurityEvents();

// Aggregate the logWriters and the in memory connection log creator
ICreateNewILog aggregateLoggerCreator = new AggregateLogWriterLogCreator(singleInMemoryConntionLogCreator, logConnectionErrors, logSecurityEvents);
```